app/controllers/api/v1/projects_controller.rb

```diff
@@ -113,10 +115,25 @@ def load_index_projects
     end
 
     def load_user_projects
-      @projects = if @current_user.id == params[:id].to_i
+      # if user is not authenticated or authenticated as some other user
+      # return only user's public projects else all
+      @projects = if @current_user.nil? || @current_user.id != params[:id].to_i
```

Comment:
`@current_user.id != params[:id].to_i` why are you excluding this? here both triggers same db query

Reply:
What if some user fetches some other user's projects? Shouldn't we only return public projects then? @tachyons

Reply:
Got it, thanks